src/main/java/net/sf/jabref/logic/importer/fileformat/BibTeXMLImporter.java

```diff
@@ -79,33 +98,184 @@ public ParserResult importDatabase(BufferedReader reader) throws IOException {
+     */
+    private <T> void parse(T entryType, Map<String, String> fields) {
+        Method[] declaredMethods = entryType.getClass().getDeclaredMethods();
+        for (Method method : declaredMethods) {
+            try {
+                if (method.getName().equals("getYear")) {
+                    putYear(fields, (XMLGregorianCalendar) method.invoke(entryType));
+                    continue;
+                } else if (method.getName().equals("getNumber")) {
+                    putNumber(fields, (BigInteger) method.invoke(entryType));
+                    continue;
+                } else if (isMethodToIgnore(method.getName())) {
+                    continue;
+                } else if (method.getName().contains("get")) {
+                    putIfValueNotNull(fields, method.getName().replace("get", ""), (String) method.invoke(entryType));
+                }
+            } catch (IllegalArgumentException | InvocationTargetException | IllegalAccessException e) {
+                LOGGER.error("Could not invoke method", e);
+            }
+        }
+    }
+
+    /**
+     * Returns whether the value of the given method name should be mapped or whether the method can be ignored.
+     *
+     * @param methodName The name of the method as String
+     * @return true if the method can be ignored, else false
+     */
+    private boolean isMethodToIgnore(String methodName) {
+        return IGNORED_METHODS.contains(methodName);
+    }
+
+    /**
+     * Inbook needs a special Treatment, because <Code>inbook.getContent()</Code> returns a list of <Code>JAXBElements</Code>.
+     * The other types have just <Code>get</Code> methods, which return the values as Strings.
+     */
+    private void parseInbook(Inbook inbook, Map<String, String> fields) {
+        List<JAXBElement<?>> content = inbook.getContent();
+        for (JAXBElement<?> element : content) {
+            String localName = element.getName().getLocalPart();
+            Object elementValue = element.getValue();
+            if (elementValue instanceof String) {
+                String value = (String) elementValue;
+                putIfValueNotNull(fields, localName, value);
+            } else if (elementValue instanceof BigInteger) {
+                BigInteger value = (BigInteger) elementValue;
+                if (FieldName.NUMBER.equals(localName)) {
+                    putNumber(fields, value);
+                }
+                if (FieldName.CHAPTER.equals(localName) && (value != null)) {
+                    fields.put(FieldName.CHAPTER, String.valueOf(value));
+                }
+            } else if (elementValue instanceof XMLGregorianCalendar) {
+                XMLGregorianCalendar value = (XMLGregorianCalendar) elementValue;
+                if (FieldName.YEAR.equals(localName)) {
+                    putYear(fields, value);
+                }
+            }
```

Comment:
I don't think that this is necessary. If there is an unexpected field, it will not be mapped by JAXB. So if I add here an else, it would be unreachable in my opinion. I've also tested this by adding a field in the ``BibTeXMLImporterTestInbookLessFields`` file by adding a field, that should not appear in the Inbook type.